app/src/lib/stores/helpers/find-upstream-remote.ts

```diff
@@ -0,0 +1,41 @@
+import { GitHubRepository } from '../../../models/github-repository'
+import { IRemote } from '../../../models/remote'
+import { parseRemote } from '../../remote-parsing'
+import { forceUnwrap } from '../../fatal-error'
+
+/** The name for a fork's upstream remote. */
+export const UpstreamRemoteName = 'upstream'
+
+/**
+ * Find the upstream remote based on the parent repository and the list of
+ * remotes.
+ */
+export function findUpstreamRemote(
```

Comment:
Might want to 📖 that this only works because GitHub is case-sensitive — the path part of a URL is case-sensitive unless the server normalizes case.

It might also be a good idea to automatically correct the remote’s case if the cases don’t match.